bionic/task_state.py

```diff
@@ -339,21 +336,39 @@ def initialize(self, bootstrap, flow_instance_uuid):
             for task_key in self.task_keys
         ]
 
+        should_persist = (
+            single_unique_element(
+                entity_def.can_persist
+                for entity_def in self.entity_defs_by_dnode.values()
+            )
+            and not self.output_would_be_missing()
+        )
+        if should_persist and bootstrap is None:
+            descriptors = [
+                task_key.dnode.to_descriptor() for task_key in self.task_keys
+            ]
+            if self.task.is_simple_lookup:
+                disable_message = """
+                applying `@persist(False)` or `@immediate` to your corresponding
+                function"""
+            else:
+                disable_message = """
+                passing `persist=False` argument when you `declare` / `assign` your
+                values"""
+            message = f"""
+            Descriptors {descriptors!r} are set to be persisted but that's
+            impossible: the cache is not available yet because core
```

Comment:
Missed a nit last time: When we say "depend on it", it sounds like "it" refers to the cache, but it really refers to the descriptors. Maybe just say "but they can't be because core bootstrap entities depend on them"?